feeds/wlan-ap/opensync/src/platform/openwrt/src/lib/target/src/vif.c

```diff
@@ -844,32 +856,36 @@ static void hs20_vif_config(struct blob_buf *b,
 
 }
 
-
 bool target_vif_config_del(const struct schema_Wifi_VIF_Config *vconf)
 {
 	struct uci_package *wireless;
 	struct uci_element *e = NULL, *tmp = NULL;
 	const char *ifname;
+	int ret=0;
 
 	vlan_del((char *)vconf->if_name);
-	uci_load(uci, "wireless", &wireless);
+	ret= uci_load(uci, "wireless", &wireless);
+	if (ret) {
+		LOGD("%s: uci_load() failed with rc %d", vconf->if_name, ret);
+		return false;
+	}
 	uci_foreach_element_safe(&wireless->sections, tmp, e) {
 		struct uci_section *s = uci_to_section(e);
-		if (strcmp(s->type, "wifi-iface"))
-			continue;
+		if ((s == NULL) || (s->type == NULL)) continue; 
+		if (strcmp(s->type, "wifi-iface")) continue;
 
 		ifname = uci_lookup_option_string( uci, s, "ifname" );
 		if (!strcmp(ifname,vconf->if_name)) {
 			uci_section_del(uci, "vif", "wireless", (char *)s->e.name, "wifi-iface");
 			break;
 		}
 	}
-	uci_commit_all(uci);
+	uci_commit(uci, &wireless, false);
```

Comment:
Is this needed considering we are setting reload_config = 1 in the next step which will commit all UCIs?

Reply:
It is required because this copy of wireless is local.   Effectively, the reload_config = 1, just tells the WM periodic_task() to call system("reload_config") to notify the rest of the world.

Reply:
Yeah I see.